WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShippingCreateLabelsViewModel.swift

```diff
@@ -510,19 +468,61 @@ private extension WooShippingCreateLabelsViewModel {
 }
 
 private extension WooShippingOriginAddress {
-    /// Converts the origin address to a `ShippingLabelAddress`.
+    /// Converts the origin address to a `WooShippingAddress`.
     ///
     /// This prepares the address for use in e.g. fetching available shipping rates or purchasing the label.
     ///
-    func toShippingLabelAddress() -> ShippingLabelAddress {
-        ShippingLabelAddress(company: company,
-                             name: fullName ?? "",
-                             phone: phone,
-                             country: country,
-                             state: state,
-                             address1: address1,
-                             address2: address2,
-                             city: city,
-                             postcode: postcode)
+    func toWooShippingAddress() -> WooShippingAddress {
+        WooShippingAddress(company: company,
+                           name: fullName,
+                           phone: phone,
+                           country: country,
+                           state: state,
+                           address1: address1,
+                           address2: address2,
+                           city: city,
+                           postcode: postcode)
+    }
+}
+
+private extension ShippingLabelAddress {
+    /// Converts the address to a `WooShippingAddress`.
+    ///
+    /// This prepares the address for use as a destination address in the shipping label.
+    ///
+    func toWooShippingAddress() -> WooShippingAddress {
+        WooShippingAddress(company: company,
+                           name: name,
+                           phone: phone,
+                           country: country,
+                           state: state,
+                           address1: address1,
+                           address2: address2,
+                           city: city,
+                           postcode: postcode)
+    }
+}
+
+private extension Address {
+    /// Converts the address to a `WooShippingAddress`.
+    ///
+    /// This prepares the address for use as a destination address in the shipping label.
+    ///
+    func toWooShippingAddress() -> WooShippingAddress {
+        // In this way we support localized name correctly,
+        // because the order is often reversed in a few Asian languages.
+        var components = PersonNameComponents()
+        components.givenName = firstName
+        components.familyName = lastName
+
+        return WooShippingAddress(company: company ?? "",
+                                  name: PersonNameComponentsFormatter.localizedString(from: components, style: .medium, options: []),
```

Comment:
I see that `Address` has helper property named `fullName` added in an extension which uses the `PersonNameComponents` to get full name from first and last names. I think we can use that. 

```suggestion
                                  name: fullName,
```

Reply:
Good catch! Updated in 1a10649.